pandas/core/array_algos/cast.py

```diff
@@ -0,0 +1,117 @@
+import inspect
+
+import numpy as np
+
+from pandas._typing import (
+    ArrayLike,
+    DtypeObj,
+)
+
+from pandas.core.dtypes.cast import (
+    astype_dt64_to_dt64tz,
+    astype_nansafe,
+)
+from pandas.core.dtypes.common import (
+    is_datetime64_dtype,
+    is_datetime64tz_dtype,
+    is_dtype_equal,
+    pandas_dtype,
+)
+from pandas.core.dtypes.dtypes import ExtensionDtype
+
+from pandas.core.arrays import ExtensionArray
+
+
+def astype_array(values: ArrayLike, dtype: DtypeObj, copy: bool = False):
```

Comment:
i think makes more sense in dtypes.cast?

Reply:
See my remark in the top post, for me either is fine

Reply:
yah lets put it next to astype_nansafe

Reply:
Moved to just below `astype_nansafe`